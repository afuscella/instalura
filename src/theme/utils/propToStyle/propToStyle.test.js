import { propToStyle } from './index';

describe('propToStype()', () => {
  describe('when propToStyle receives an sample argument', () => {
    test('and it is a string, should convert to style', () => {
      const propToStyleResult = propToStyle('textAlign');
      const componentProps = { textAlign: 'center' };
      const styleResult = propToStyleResult(componentProps);

      expect(styleResult).toEqual({ textAlign: 'center' });
    });

    test('and it is a number, should convert to style', () => {
      const propToStyleResult = propToStyle('flex');
      const componentProps = { flex: 1 };
      const styleResult = propToStyleResult(componentProps);

      expect(styleResult).toEqual({ flex: 1 });
    });
  });

  describe('when propToStyle receives an argument with breakpoint', () => {
    test('', () => {

    });
  });
});
