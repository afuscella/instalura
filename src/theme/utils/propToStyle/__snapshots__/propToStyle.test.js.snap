// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`propToStype() when propToStyle receives an argument with breakpoint render only a single large breakpoint resolution 1`] = `
Array [
  "@media only screen and (min-width:",
  "1200",
  "px){",
  "text-align: center;",
  ";}",
]
`;

exports[`propToStype() when propToStyle receives an argument with breakpoint render only a single small breakpoint resolution 1`] = `
Array [
  "@media only screen and (min-width:",
  "0",
  "px){",
  "text-align: center;",
  ";}",
]
`;

exports[`propToStype() when propToStyle receives an argument with breakpoints render multiple breakpoints resolutions 1`] = `
Array [
  "@media only screen and (min-width:",
  "480",
  "px){",
  "text-align: center;",
  ";}",
  "@media only screen and (min-width:",
  "768",
  "px){",
  "text-align: left;",
  ";}",
  "@media only screen and (min-width:",
  "992",
  "px){",
  "text-align: center;",
  ";}",
]
`;
